### ISSUE TITLE
Library team

### ISSUE THREAD

**Issue by AndrewBelt** [VCVRack/library#352]
Participating in the Library team is the simplest of the four. Requires knowledge of submitting PRs and editing JSON in the `manifests/` directory.

## Tasks

- Keep plugin manifests correct and up-to-date.
- Handle issues opened by plugin developers who want to add/update their plugin.
- Seek new plugins/updates when developers don't notify us.

## Workflow

Begin by checking this repo's issues for requests from plugin developers.

If someone requests an information change, simply make a commit (and send a PR if you don't yet have VCV community access) with the change to the manifests following the manifest format at https://github.com/VCVRack/community#library-team, and someone with write access will accept it after a quick review. You may combine multiple manifest updates in one PR.
Do not modify or add the `"status"` or `"latestVersion"` property.

If someone requests an update to their build, leave that to the Review team (or *become* part of the build team by reading the instructions #353).

If someone requests a new plugin to be added, add a new JSON file to `manifests/` with all known information. Do not add the `"status"` or `"latestVersion"` property.

Finally, send a pull request, and someone with write access will review and accept it.
After accepted, notify the plugin developer by commenting "Updated information" in the original plugin's thread.

I think that's it, pretty simple. I may add JSON validation scripts that you should run before committing, but I'll wait until we've had a case of invalid JSON. Expecting all plugin developers to write valid JSON (like we did in during Rack 0.5) is impossible, but I can assume Library Team volunteers will do most everything correctly since you'll be more familiar with the process.

## Members

- @cschol
- @phdsg

**Comment by AndrewBelt** [comment]
I'll use this issue https://github.com/VCVRack/community/issues/356 as a practice for someone on the Library team. Typically, each plugin developer will create an issue with the title = plugin slug (I'll personally change it if not) where they'll communicate with the Library team upon updates.

@MarcBoule has provided a scratch JSON file. Usually developers won't do that. Some things the Library team should note about this particular example:
- Remove duplicate URLs. In this case, `websiteUrl` should be removed since a GitHub URL is not their custom website. `manualUrl` should link directly to their README, if it is helpful to end users
- Status should be removed. I'll figure out what to do with that later.

**Comment by MarcBoule** [comment]
DISCUSSION: Just to understand better the proposed method, is it your intention that each dev's issue, titled with his or her slug as mentionned above, remain persistent (i.e. always open), and thereby serves as the communication channel between the dev and the Library team? In this manner a request for an update of the repo's version of our plugin, for example, would be a new comment in the plugis "issue" of that dev, instead of a new issue each time?

**Comment by AndrewBelt** [comment]
Yes, keep the same issue for each plugin.
If the issue creator closes the issue, can they reopen it? If not, just keep it open indefinitely.

**Comment by cschol** [comment]
A few questions:
- Should the manifest .json files always contain **all** keys, e.g. `contactEmail` or `donateUrl`, even if the value is not known or used (e.g. no contact email provided or no donations)?
- Should we tag the plugin developer in the PR for updating the manifest to review?

**Comment by n6smith** [comment]
Note: Both links in initial post are non functioning 404s.

**Comment by AndrewBelt** [comment]
@n6smith Thanks, switched to master branch.

**Comment by cschol** [comment]
@AndrewBelt Would you prefer one PR per plugin update or is it OK to combine multiple plugins in one PR?

**Comment by AndrewBelt** [comment]
- An update to the build requires that the user change the version somehow, so Rack knows that an update is available. It checks `oldVersion != newVersion` as a criteria for requesting an update.
- I'll type something to answer these questions in the #353 issue

**Comment by phdsg** [comment]
case: #398
dev requested addtition. manifest was created and accepted. thread says "updated information"...
is this the point where the review team picks up and creates the submodule?

**Comment by AndrewBelt** [comment]
The Library and Review teams can work in parallel. If you want to act on behalf of both teams, you can combine your commit as long as you follow both workflows (e.g. acknowledge that the repo has been reviewed).

**Comment by phdsg** [comment]
creators can reopen their issues unless you closed them, right?
i think it'd be good to encourage devs to close their issue unless there's something to do...
just for readability in the issue tracker.

**Comment by AndrewBelt** [comment]
@phdsg Yes, that's what the test at #393 concluded.
I'm writing a better documentation for plugin developers.

**Comment by phdsg** [comment]
do you handle the free closed-source plugins like #376 and others?

**Comment by AndrewBelt** [comment]
I haven't thought of a procedure for that yet.

**Comment by AndrewBelt** [comment]
Thanks for the help so far guys! To organize team members, state your name and I'll add you to the first post as a member.

**Comment by cschol** [comment]
Great work! I am going to catch up and mark all 238971231 GitHub notifications as read. If you need help with any specific plugin let me know.

**Comment by cschol** [comment]
I used to be able to add tags to issues filed, e.g. plugin, but can't anymore. @AndrewBelt Are you taking care of that?

**Comment by AndrewBelt** [comment]
I've disabled everyone's write access to this repo until later. Still need to make sure everything will be work properly when PRs are no longer needed.

**Comment by cschol** [comment]
Ah, cool. No problem.

**Comment by AndrewBelt** [comment]
@phdsg Good question. Yes, I'll handle everything for the open-source VCV plugins.

**Comment by AndrewBelt** [comment]
@cschol @phdsg I've given you commit access so we can finally skip PRs.

If someone would like to join the Library Team, send a few correct PRs following the **Workflow** in the first post, and I'll also give you commit access.

**Comment by phdsg** [comment]
i suppose
> - Seek new plugins/updates when developers don't notify us.

is just to open an "invitation" at their repo and wait for them to open their channel on the community tracker?

**Comment by AndrewBelt** [comment]
@phdsg Sure, you can open their plugin thread for them (and "@" their username!) Although, if a volunteer is capable of researching for new plugin information *and* can follow the Library workflow, it's perfectly fine to bypass the plugin thread entirely. There are plenty of plugins in `manifests/` without plugin threads that could use updating.

**Comment by phdsg** [comment]
i probably won't add anybody's repo who doesn't request it.
inviting unlisted devs seems to be a perfect entry level task tho.

**Comment by phdsg** [comment]
also, could devs still close issues opened by one of the team members?

**Comment by AndrewBelt** [comment]
I haven't tested whether the issue OP can close an issue reopened by a collaborator, but I know that they can't reopen an issue closed by a collaborator.

**Comment by AndrewBelt** [comment]
I've been informed that many versions are incorrect (e.g. < 0.6) on https://vcvrack.com/plugins.html for plugins not available in the build system.

So, if `"status"` is *not* `"available"`, you are free to set `"latestVersion"`.

**Comment by phdsg** [comment]
did a few of them...

it'd be cool to find a way to make more people participate here instead of going to places like [this](http://www.switchedonrack.com/plugins-currently-not-in-the-plugin-manager.html) or on fb

**Comment by AndrewBelt** [comment]
@phdsg Should I make a quick post on the Facebook user group to request 2-3 more Library Team members?

**Comment by AndrewBelt** [comment]
We need to add this to the workflow: When a developer requests a change on their plugin's issue, a repo maintainer should open it, and then close it when the change has been satisfied. Unfortunately when a maintainer closes an issue, the OP can't reopen it, so maintainer will have to open/close all issues by watching the issues. But some additional questions:

- If I close an issue, can a maintainer reopen it?
- If I open an issue, can a maintainer close it?

**Comment by phdsg** [comment]
works fine.

**Comment by AndrewBelt** [comment]
Great! When I'm at a computer I'll close all the issues that have no withstanding changes. You're welcome to get started on that before me though.

**Comment by AndrewBelt** [comment]
Done. Here's a handy link to [all open plugin issues sorted by recently updated](https://github.com/VCVRack/community/issues?q=is%3Aissue+is%3Aopen+label%3Aplugin+sort%3Aupdated-desc). This would be a good bookmark.

**Comment by AndrewBelt** [comment]
All of the currently open ones should probably be closed, but it wasn't immediately clear the their status is.

**Comment by AndrewBelt** [comment]
Suggestion: Previously I requested to avoid redundancy and not include a `pluginUrl` if it was the same link to `sourceUrl` or `manualUrl`. Now, let's try to include a `pluginUrl` on as many plugins as possible, given that the URL is targeted toward users. For example, LOGinstruments has a `sourceUrl` of https://github.com/leopard86/LOGinstruments, but let's now add https://github.com/leopard86/LOGinstruments/blob/master/README.md as the `pluginUrl`.

**Comment by cschol** [comment]
Stats:
- Total plugins: 137
- pluginUrl exists: 61
- no pluginUrl, but manualUrl exists: 69
- no pluginUrl, no manualUrl, potentially built from sourceUrl: 7

**built from sourceUrl** means: `sourceUrl + "blob/<branch name>/README.md"`, **IF** `README.md` exists on a branch, e.g. `master`.

I can script the update of the manifests and push an update to the repo to set `pluginUrl` for all plugins. Thoughts?

**Comment by AndrewBelt** [comment]
It isn't necessarily true that the README is the best URL. Sometimes the github wiki, sometimes a new website was added. Sometimes the README isn't a good URL for users, so it shouldn't be the `pluginUrl`. Since it's only 69 without, I'd just do it on a case-by-case.

**Comment by cschol** [comment]
For the 69 the `pluginUrl` would be the `manualUrl` (whatever it is set to) since it exists. The remaining ones without a clear choice are only 7.

**Comment by AndrewBelt** [comment]
Okay, you can script whatever you need, but it would be good to review the changes afterwards.

**Comment by cschol** [comment]
Sounds good. I'll just do a PR.

**Comment by AndrewBelt** [comment]
You can commit directly.

**Comment by AndrewBelt** [comment]
Rack v1 will require all plugins (except Core) to supply a `plugin.json` file in its root. (Example at https://github.com/VCVRack/Template/blob/7c28f8d276e51d9d2c8e8de639e6823648c97e8c/plugin.json.) There are many great things we can do with this metadata. The file will be written by the plugin author rather than the Library team, so this reduces the team's responsibilities to zero, or rather transfers the responsibility to the Review team to review the metadata according to guidelines.

**Comment by netboy3** [comment]
I'm the new maintainer of MSM and am getting ready to push 1.0.0 to the library:

- Should I use the old "MSM" issue to post a request or open a new "MSM" issue?
- Would posting the 1.0.0 overwrite/remove the 0.6.51 entry in the plugin library?

**Comment by AndrewBelt** [comment]
You can use the same issue. Yes, since you'll be using the MSM slug, it will overwrite the old manifest and thus the entry on the VCV Library website.

**Comment by RemiCollin** [comment]
Do you need help with the library repo ? I can help on managing github issues / do code review if you want.

**Comment by AndrewBelt** [comment]
Closing since this issue's purpose has been served, and the [VCV Community forum](https://community.vcvrack.com/) is a more organized place for discussions.
